src/onegov/api/models.py

```diff
@@ -81,9 +81,8 @@ def __init__(self, app: 'Framework', endpoint: str, id: str):
         self.id = id
 
     @property
-    def api_endpoint(self) -> 'ApiEndpoint[_M] | None':
+    def api_endpoint(self) -> 'ApiEndpoint[Any] | None':
```

Comment:
```suggestion
    def api_endpoint(self) -> 'ApiEndpoint[_M] | None':
```

Now that `ApiEndpointItem` is itself generic you can actually return `ApiEndpoint[_M]` here, the reason it was complaining before was that there were two different versions of `_M` involved, an unbound one and a bound one, which are incompatible.

Some other type checkers such as pyright help visualize this issue a bit better by adding a unique numeric suffix to each type variable instance, so it's easier to distinguish different instances of the same type variable in error messages.

Reply:
Ah _that_ was the reason. I was getting confused because of this nonsensical error message, something like expected type `_M`, got type `_M`